gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/ddl/catalog/GfxdSystemProcedures.java

```diff
@@ -2037,6 +2039,37 @@ public static String GET_NATIVE_NANOTIMER_TYPE() {
   }
 
   /**
+   * Repair Snappy catalog (Hive MetaStore and data dictionary) by removing
+   * inconsistent entries in the catalog.
+   * @throws SQLException
+   * @throws StandardException
+   */
+  public static void REPAIR_CATALOG() throws SQLException, StandardException {
+    if (GemFireXDUtils.TraceExecute) {
+      SanityManager.DEBUG_PRINT(GfxdConstants.TRACE_EXECUTION,
+          "in procedure REPAIR_CATALOG()");
+    }
+    final boolean isLead = GemFireXDUtils.getGfxdAdvisor().getMyProfile().hasSparkURL();
```

Comment:
This will always hit a server. So always publish will happen. But we can add a comment and leave like this.